File: data structures and algorithms/cpp/05_009.cpp
class BrowserHistory {
private:
    std::vector<std::string> history; // Our stack
    
public:
    void visit_page(const std::string& url) {
        history.push_back(url);
    }
    
    std::string go_back() {
        if (!history.empty()) {
            std::string page = history.back();
            history.pop_back();
            return page;
        }
        return "";
    }
};

File: data structures and algorithms/cs/05_009.cs
using System;
using System.Collections.Generic;

public class MainClass {
   public static void Main(string[] args) {
      // Main method code goes here
   }
   
   public class BrowserHistory {
      // Our stack
      private Stack<string> history;
      
      public BrowserHistory() {
         this.history = new Stack<string>();
      }
      
      public void VisitPage(string url) {
         this.history.Push(url);
      }
      
      public string GoBack() {
         if (this.history.Count > 0) {
            return this.history.Pop();
         }
         return null;
      }
   }
}